packages/app/src/DashboardFiltersEditModal.tsx

```diff
@@ -0,0 +1,284 @@
+import { useEffect, useRef, useState } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { TableConnection } from '@hyperdx/common-utils/dist/metadata';
+import {
+  DashboardFilter,
+  MetricsDataType,
+  SourceKind,
+} from '@hyperdx/common-utils/dist/types';
+import {
+  Button,
+  Flex,
+  Group,
+  Input,
+  Modal,
+  Paper,
+  Radio,
+  Stack,
+  Text,
+  TextInput,
+  UnstyledButton,
+} from '@mantine/core';
+
+import SourceSchemaPreview from './components/SourceSchemaPreview';
+import { SourceSelectControlled } from './components/SourceSelect';
+import { SQLInlineEditorControlled } from './components/SQLInlineEditor';
+import { useSource } from './source';
+import { getMetricTableName } from './utils';
+
+interface DashboardFilterEditFormProps {
+  filter: DashboardFilter;
+  onSaveFilter: (definition: DashboardFilter) => void;
+  onRemoveFilter: (id: string) => void;
+  parentRef?: HTMLElement | null;
+}
+
+const DashboardFilterEditForm = ({
+  filter,
+  onSaveFilter,
+  onRemoveFilter,
+  parentRef,
+}: DashboardFilterEditFormProps) => {
+  const { handleSubmit, register, formState, control, watch, reset } =
+    useForm<DashboardFilter>({
+      defaultValues: filter,
+    });
+
+  useEffect(() => {
+    reset(filter);
+  }, [filter, reset]);
+
+  const onSubmit = (data: DashboardFilter) => {
```

Comment:
If this is all we're doing, we could just do `<form onSubmit={handleSubmit(onSaveFilter)}>`